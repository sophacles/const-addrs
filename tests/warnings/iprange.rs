use const_addrs::iprange;

fn main() {
    let _ = iprange!("2001:db8::32:ffff", "2001:db8::32:23");
    let _ = iprange!("10.128.0.0", "10.0.0.0");

    //failure so the output is compared
    let _ = iprange!("10128.0.0", "10.0.0.0");
}
