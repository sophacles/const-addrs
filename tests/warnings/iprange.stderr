warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/iprange.rs:4:22
  |
4 |     let _ = iprange!("2001:db8::32:ffff", "2001:db8::32:23");
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

warning: This range will yeild no values: IP1 >= IP2
 --> tests/warnings/iprange.rs:5:22
  |
5 |     let _ = iprange!("10.128.0.0", "10.0.0.0");
  |                      ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: This iterator requires the lower IP value first in it's arguments list.

error: invalid IP address syntax
 --> tests/warnings/iprange.rs:8:22
  |
8 |     let _ = iprange!("10128.0.0", "10.0.0.0");
  |                      ^^^^^^^^^^^
